libethereum/ExtVM.cpp

```diff
@@ -45,13 +55,23 @@ h160 ExtVM::create(u256 _endowment, u256& io_gas, bytesConstRef _code, OnOpFunc
 	// Increment associated nonce for sender.
 	m_s.noteSending(myAddress);
 
-	Executive e(m_s, lastHashes, depth + 1);
-	if (!e.create(myAddress, _endowment, gasPrice, io_gas, _code, origin))
+	m_callStack.emplace(m_s, lastHashes, depth + 1);
+	try
+	{
+		Executive& e = m_callStack.top();
+		if (!e.create(myAddress, _endowment, gasPrice, io_gas, _code, origin))
+		{
+			e.go(_onOp);
+			e.accrueSubState(sub);
+		}
+		io_gas = e.endGas();
+		m_callStack.pop();
+		return e.newAddress();
+	}
+	catch(...)
 	{
-		e.go(_onOp);
-		e.accrueSubState(sub);
+		m_callStack.pop();
```

Comment:
Why not use a scope guard here? If `e.newAddress` throws, the stack is popped twice.